### PR TITLE
Feature/robot  pmtalloc pom

### PR DESCRIPTION
**Pull request by skristem**
Refactoring Payment Allocations existing tests to Page Object Model
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot [hunk 1]
```diff
@@ -12,49 +15,72 @@ Suite Teardown  Run keywords
...
+    ...                                    Add a new Allocation to another GAU and verify that the Allocation is split between default and new GAUs.
+    ...                                    Go to Opportunity and verify that same allocation is reflected under GAU allocations related list. 
+    [tags]                                 W-039821            feature:Payment Allocations
+    Go To Page                             Detail
+    ...                                    Opportunity
+    ...                                    object_id=${data}[contact1_opportunity][Id]
+    Select Tab                             Related
+    Load Related List                      GAU Allocations
+    Click Link With Text                   &{payment1}[Name]    
     Select Window
```
Comment (sgovindankutty) [inline robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: You are not passing value of the window? is this intentional?

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot [hunk 2]
```diff
@@ -12,49 +15,72 @@ Suite Teardown  Run keywords
...
+    ...                                    Open Opportunity and Manage Allocations from GAU Allocations related list.
+    ...                                    Add a new Allocation to another GAU and verify that the Allocation is split between default and new GAU.
+    ...                                    Open Payments from related list and verify that same allocation is reflected under Payment Allocations related list.
+    [tags]                                 W-039821                   feature:Payment Allocations
+    Go To Page                             Detail
+    ...                                    Opportunity
+    ...                                    object_id=${data}[contact2_opportunity][Id]
+    Select Tab                             Related
+    Click Special Related List Button      GAU Allocations            Manage Allocations
+    Wait For Page Object                   Custom                     ManageAllocations
```
Comment (sgovindankutty) [inline robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: Are Manage Allocations in line 70 & 71 the same or different?

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot [hunk 3]
```diff
@@ -64,55 +90,50 @@ Update GAU Allocations and Verify Payment Allocations Sync
 Enable Payment Allocations
     &{def_gau} =  API Create GAU    Name=default gau
     Set Global Variable     &{def_gau}    &{def_gau}
```
Comment (sgovindankutty) [inline robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: Why is &{def_gau} called twice while setting the global variables? Is there any reason behind it?

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot (hunk 1 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: yes, it selects the active window by default

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot [hunk 4]
```diff
@@ -64,55 +90,50 @@ Update GAU Allocations and Verify Payment Allocations Sync
...
-    ...    npe01__Opportunity__c=&{opportunity}[Id]
+    ...    npe01__Opportunity__c=${data}[contact2_opportunity][Id]
     &{id} =     Get From List  ${records}  0
-    &{payment} =     Salesforce Get  npe01__OppPayment__c  &{id}[Id]  
-    Set suite variable    &{opportunity}
-    Set suite variable    &{payment} 
\ No newline at end of file
+    &{payment2} =     Salesforce Get  npe01__OppPayment__c  &{id}[Id]  
+    Set suite variable    &{payment2} 
+    Log   ${data}
```
Comment (rjanjanam) [inline robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: This log, may be you forgot to take out?

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot (hunk 2 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: line 70 is button name, 71 is page object name

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot (hunk 4 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: yes thanks for catching

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot (hunk 3 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: as specified on the call used it bcoz of syntax but converted to Suite Variable as not needed as global

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot (hunk 4 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: removed

File: robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot (hunk 3 above)
Comment (sgovindankutty) [reply robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot]: We went over this in call.

**Pull request merged**